src/tools/rbd/action/DiskUsage.cc

```diff
@@ -180,9 +180,7 @@ static int do_disk_usage(librbd::RBD &rbd, librados::IoCtx &io_ctx,
           goto out;
         }
 
-        if (snapname != NULL) {
-          total_prov += snap->size;
-        }
+        total_prov += snap->size;
```

Comment:
Thanks, @dillaman
It wouldn't double the provisioned total when 'rbd du' a image.
In fact, it's absolutely right that it calculate the original image size in line 206, but the line 184 only works when 'rbd du' a snapshot, so I modified it.  
In my opinion, this provisioned total is about the volume in one pool, so it must include the image size and snapshot size.
Now it would add the snapshot's size when 'rbd du' not only a image, but also a snapshot, because input var snapname is NULL when 'rbd  du' a image.